hudi-hadoop-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -102,8 +102,11 @@ private FSDataOutputStream getOutputStream() throws IOException {
       boolean created = false;
       while (!created) {
         try {
+          if (storage.exists(logFile.getPath())) {
```

Comment:
if we do not do this, we end up trying to create the same log file as previous version. and then hit FileAlreadyExistsException. and then we go ahead and create a new log file. 
but by this time, two marker files are created, 1st one pointing to the previous committed log file. and the 2nd one referring to the log file created in the current inflight delta commit. 

and during marker reconciliation, we end up deleting the 1st log file which is part of committed delta commit :(

Reply:
illustration
C1: 
basefile1 with commit time 001. 
C2: 
lf1 (.d4c17444-7d2c-4608-8ad0-fd20efedf1c8-0_001.log.1_1-0-1)

C3: 
supposed to add lf2 with name (.d4c17444-7d2c-4608-8ad0-fd20efedf1c8-0_001.log.2_1-0-1) // check for version no.
, but when we reach L105 above, log file name refers to the latest known log file. and so it refers to .d4c17444-7d2c-4608-8ad0-fd20efedf1c8-0_001.log.1_1-0-1. We go ahead and try creating the file (which creates the marker),
but then log file creation fails. 

and we rollover and then end up creating marker and actual data file for .d4c17444-7d2c-4608-8ad0-fd20efedf1c8-0_001.log.2_1-0-1


When we reach marker reconciliation step for this dc 003, we deduce that log file w/ version 2 is spurious and delete it. and commit 003.

Reply:
Discussed offline, this isn't extra overhead over what we already had in 0.x. But, we need to understand two things:
1. Why was it removed?
2. Is it really needed? It looks like an idempotency guard against fsview/task retries. But, let's revisit later if we can optimize the log format writer flow.